R/glify-helpers.R

```diff
@@ -169,3 +169,42 @@ glifyDependenciesFl = function() {
     )
   )
 }
+
+
+
+json_funccall <- function() {
+  json_parser <- getOption("leafgl_json_parser", "jsonify")  # Default to jsonify
+  if (json_parser == "yyjsonr") {
+    yyjsonr::write_json_str
+  } else {
+    jsonify::to_json
+  }
+}
+convert_to_json <- function(data, ...) {
+  json_parser <- getOption("leafgl_json_parser", "jsonify")  # Default to jsonify
+  if (json_parser == "yyjsonr") {
+    print("I am using yyjsonr")
+    json_data <- yyjsonr::write_json_str(data, ...)
+    class(json_data) <- "json"
+  } else {
+    print("I am using jsonify")
+    json_data <- jsonify::to_json(data, ...)
+  }
+  return(json_data)
+}
+
+## Not used as its not faster - Needs geometries to be the last column and be named geometry
+yyjsonr_2_geojson <- function(sfdata) {
+  # sfdata <- data
+  geom <- st_geometry(sfdata)
+  sfdata <- st_drop_geometry(sfdata)
+  sfdata$geometry <- geom
+  sfdata <- sf::st_as_sf(sfdata)
+  json_data <- yyjsonr::write_json_str(sfdata, digits=4)
```

Comment:
You are correct! Apparently I missed that function. I will add it.
I think we only have to change the class of the `yyjsonr::write_geojson_str` to "json", then we shouldnt need the JSON.parse on the JS side. 

a little benchmark 👀 looks promising!
```
mc <- microbenchmark::microbenchmark(times = 10,
  yyjsonr = yyjsonr::write_geojson_str(mapview::breweries),
  geojsonsf = geojsonsf::sf_geojson(mapview::breweries)
); mc
```
```
Unit: microseconds
      expr      min       lq      mean   median       uq      max neval
   yyjsonr  203.701  208.501  218.7708  214.651  225.600  248.902    10
 geojsonsf 1654.702 1662.301 1726.0412 1675.801 1763.402 1951.501    10
```

Reply:
I just pushed the changes. There is now another option `leafgl_geojson_parser` which defaults to `geojsonsf`.
Should we make `yyjsonr` the default?

Reply:
Honestly, I don't see why we would need such options for the parser? I don't think anyone will care about the internals. I would vote for simply using `yyjsonr`. Or am I missing something?

Reply:
I thought it would be easier to test out other Json packages as they emerge.
But I guess you're right, not many would care about this and probably wouldnt change the options to yyjsonr.

I'll fix it. Also currently the arguments wouldnt be passed correctly.

Reply:
I switched completely to yyjonsr and removed jsonify and geojsonsf from the Imports.

I had to fix the example where `popup=FALSE`. Jsonify returned `{}` and yyjsonr returns `[]`, but I guess we should catch that in R already (and set to NULL), as currently it will throw this console error:

`Uncaught TypeError: Cannot read properties of undefined (reading 'toString')`


```
leaflet() %>% addTiles() %>%
    addGlPolygons(data = gadm, popup = FALSE, opacity = 1)
```

Reply:
I just tried if this would fix/improve #72, but the example throws an error now. The current CRAN version works.
```
RangeError: Invalid array length
    at Array.push (<anonymous>)
    at g.resetVertices (glify-browser.js:9:84546)
    at g.render (glify-browser.js:9:83337)
    at new g (glify-browser.js:9:83191)
    at M.lines (glify-browser.js:9:101385)
```

Interestingly, if I subset the data, it works with the new version aswell.
It works with `d_[1:400000,]` or `d_[400000:nrow(dat),]` for example.. So it can render the whole data, just not at once.. ?

Apparently the breaking change comes from the new Leaflet.Glify version. I thought it's maybe because of yyjsonr, so I re-installed the current github version but the error remains.